daemon/daemon.go

```diff
@@ -1602,24 +1602,37 @@ func (daemon *Daemon) initDiscovery(config *Config) error {
 func (daemon *Daemon) Reload(config *Config) error {
 	daemon.configStore.reloadLock.Lock()
 	defer daemon.configStore.reloadLock.Unlock()
-	daemon.configStore.Labels = config.Labels
+	if config.IsValueSet("label") {
+		daemon.configStore.Labels = config.Labels
+	}
+	if config.IsValueSet("debug") {
```

Comment:
This is not necessary, but I'm okay with adding it for consistency, that field is only in use when the server starts.